practices/what_is_rest_api

```diff
@@ -0,0 +1,79 @@
+
+###How to Implement RESTful API
+
+![How RESTful API Works](/files/how_restful_api_works.png)
+
+A RESTful API uses commands to obtain resources. In REST, each resource is identified by a unique URI (Uniform Resource Identifier) and can be manipulated using standard HTTP methods as defined by the RFC 2616 protocol, such as:  
+
+* GET to retrieve a resource;
+* PUT to change the state of or update a resource, which can be an object, file or block;
+* POST to create that resource; and
+* DELETE to remove it.
+
+It is based on the idea that the REST APIs are stateless, meaning that calls can be made independently of one another, and each call contains all of the data necessary to complete itself successfully
+
+**The core principles of RESTful API**
+The principles of REST are centered around a few core concepts, which include:
+
+*  **Stateless communication**: RESTful APIs should be stateless, meaning that no client context should be stored on the server between requests. Each request should contain all the necessary information to complete that request, including authentication credentials if required.
+
+*  **Resource identification**: Resources should be identified using unique URIs (Uniform Resource Identifiers) that are included in the request. Each resource should have a unique URI, and requests should use these URIs to manipulate the resource.
+*  **Uniform interface**: A uniform interface should be used to access resources. This means that HTTP methods should be used to access resources, and the same HTTP method should have the same effect on all resources.
+*  **Representation of resources**: Resources should be represented in a format that is easily consumable by the client. This can include JSON, XML, or other formats.
+*  **Hypermedia as the engine of application state (HATEOAS)**: This principle states that the client should be able to navigate the API using hyperlinks included in the response. This allows the API to evolve without breaking existing clients.
+
+**Advantages of REST API**
+
+*  **Scalability**: RESTful APIs are highly scalable due to their stateless nature, which allows for easy scaling of resources.
+*  **Flexibility**: RESTful APIs are flexible in terms of the format of data, which allows for easy integration with other web-based systems.
+*  **Easy to understand**: RESTful APIs are easy to understand due to their simplicity and uniform interface.
+*  **Easy to develop**: RESTful APIs are easy to develop due to their uniform interface and stateless nature, which reduces the complexity of the application.
+*  **Platform independence**: RESTful APIs can be used with any programming language and platform, making them highly interoperable.
+
```

Comment:
Please, add missing sections "Common Pitfalls of the REST API" and "Resources for the REST API" according to https://github.com/DXHeroes/knowledge-base-content/blob/master/CONTRIBUTING.md#content-composition

Reply:
Completed